src/main/app/cypress/integration/haku.spec.js

```diff
@@ -10,34 +10,36 @@ describe('Haku', () => {
   it('Koulutustyyppi checkboxes should work hierarchically', () => {
     cy.visit('/fi/haku/auto');
 
-    cy.findByRole('checkbox', { name: /Ammatillinen koulutus/ }).as(
-      'AmmatillinenKoulutus'
-    );
-    cy.findByRole('checkbox', { name: /Ammatillinen perustutkinto/ }).as(
-      'AmmatillinenPerustutkinto'
-    );
-    cy.findByRole('checkbox', { name: /Ammattitutkinto/ }).as('Ammattitutkinto');
-    cy.findByRole('checkbox', { name: /Erikoisammattitutkinto/ }).as(
-      'Erikoisammattitutkinto'
-    );
-
-    cy.get('@AmmatillinenKoulutus').check();
-    cy.get('@AmmatillinenPerustutkinto').should('be.checked');
-    cy.get('@Ammattitutkinto').should('be.checked');
-    cy.get('@Erikoisammattitutkinto').should('be.checked');
-
-    cy.get('@AmmatillinenKoulutus').uncheck();
-    cy.get('@AmmatillinenPerustutkinto').should('not.be.checked');
-    cy.get('@Ammattitutkinto').should('not.be.checked');
-    cy.get('@Erikoisammattitutkinto').should('not.be.checked');
-
-    cy.get('@AmmatillinenPerustutkinto').check();
-    cy.get('@AmmatillinenKoulutus').should('have.attr', 'data-indeterminate');
-
-    cy.get('@Ammattitutkinto').check();
-    cy.get('@Erikoisammattitutkinto').check();
-
-    cy.get('@AmmatillinenKoulutus').should('be.checked');
+    cy.get('[data-cy=koulutustyyppi-filter]').within(() => {
```

Comment:
Nämä olis parempi laittaa cypress-testing-libraryn kautta. Sille pitää ensin konffia, että mikä tuo käytetty testi-id-attribuutti on. Kouta-ui:ssa näin: https://github.com/Opetushallitus/kouta-ui/blob/master/src/main/app/cypress/support/index.js#L5

Erona siis vaan, että konfo-ui:ssa on käytetty `data-cy`-attributtia.

Tuon konffin jälkeen jälkeen voi tehdä näin:
```suggestion
    cy.findByTestId('koulutustyyppi-filter').within(() => {
```

Reply:
Tää on hyvä :+1: